### ISSUE TITLE
Can we use Quantity.Scale to prevent certain operations?

### ISSUE THREAD

**Issue by keilw** [unitsofmeasurement/unit-api#167]
There was a question at OOP (@filipvanlaenen will remember) about preventing e.g. the `multiplication` of `CELSIUS`. 
Is a certain `Quantity.Scale` able to control that, or would we need a `LevelOfMeasurement` enum after all? https://en.wikipedia.org/wiki/Level_of_measurement#Mathematical_operations tells, that multiplication is only allowed on values at `RATIO` level, so can `RELATIVE` also be used to throw an exception e.g. for `Quantity.multiply()` or do we have to get the level back in all relevant places?

**Comment by desruisseaux** [comment]
Said otherwise: Wikipedia said that `CELSIUS` as an absolute measurement can not be multiplied _as-is_, but it does not said that we are not allowed to convert to something else (`KELVIN`) for enabling multiplication.

**Comment by keilw** [comment]
There should be some `throws` or similar description in JavaDoc, like in `Unit.root()`. 
Having defined that `enum` in `Quantity` only, what about `Unit.multiply(Unit)`?
Is `CELSIUS.multiply(CELSIUS)` OK? I guess it might because the result would be another unit, but just to cover all operations.

https://www.wikihow.com/Convert-Between-Fahrenheit,-Celsius,-and-Kelvin mentions conversions that involve `Quantity.multiply(1.8)` as part of converting a `CELSIUS` temperature to `FAHRENHEIT`. 
So if we do prevent it for any scale or other precondition it should be well-documented to avoid surprised or angry users.

**Comment by desruisseaux** [comment]
I don't think we should throw exception. `Quantity.multiply(Quantity)` is a valid operation even if the quantities are absolute measurement in `CELSIUS`, providing that the quantities are converted to `KELVIN` before multiplication. The rule is that we can not multiply `CELSIUS` _as-is_. There is no rule saying that we can not multiply the value **after conversion** to a ratio unit.

**Comment by keilw** [comment]
So what is the preferred runtime behavior in this case? 
Same as before? If so, then I guess we can close this ticket, if no action is required.

So far https://github.com/unitsofmeasurement/indriya/issues/128 in the RI is the only issue that looks like it may use the scale.

**Comment by desruisseaux** [comment]
I don't know what is "same as before". The scale is used for determining how to convert to `KELVIN` before the multiplication:

1. If absolute: convert 2°C as 275.15 K, then multiply.
2. If difference: convert 2°C as 2 K, then multiply.

Implementation is free to convert the result back to °C if they wish. My proposal is to keep the result of 1 in kelvin for less surprising behavior.

**Comment by keilw** [comment]
Ok but it would not be prevented e.g. by throwing an `ArithmeticException` or similar. 
That was asked by an OOP audience member. Unless we consider this crucial right now,  I would rather not reintroduce the `LevelOfMeasurement` (which would help to prevent multiplication for some levels) at least not in 2.0. Unless there is a demand for this?

**Comment by desruisseaux** [comment]
I see no need for `LevelOfMeasurement` and no need to prevent multiplication for some levels. What is forbidden is doing multiplication **without prior conversion** for some levels.
